cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -114,10 +116,14 @@ void SparsePoseGraph::AddScan(
     submap_states_.back().trajectory = submaps;
     CHECK_EQ(submap_states_.size(), submap_indices_.size());
   }
+  // ... One call before the first of insertion_submaps was finished
   const mapping::Submap* const finished_submap =
       insertion_submaps.front()->finished_probability_grid != nullptr
           ? insertion_submaps.front()
           : nullptr;
+  // by providing finished_submap (!=nullptr) to ComputeConstraintsForScan
```

Comment:
Actually I think `ComputeConstraintsForScan` should not call `ComputeConstraintsForOldScans`. (To make my point  more clear: Otherwise the first one should be called `ComputeConstraintsForThisScanAndOldScansIfSubmapFinished`. This might sound sarcastic, but I honestly just meant to describe the problem I had with understanding the code :) ) I'm not sure if there is a more elegant solution to this, so for now I figured pointing this out shortly before `ComputeConstraintsForScan` is used might be sufficient.